src/components/blocks/AccordionBlock.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react'
+import type { Template } from 'tinacms'
+import { Accordion } from '@mantine/core'
+import { PageBlocksAccordionblock } from '~tina/__generated__/types'
+import { Icon as Iconify } from '@iconify/react'
+
+export const AccordionBlock = ({ data }: { data: PageBlocksAccordionblock }) => {
+	const [activeIndexes, setActiveIndexes] = React.useState<number[]>([])
+
+	const toggleIndex = (index: number) => {
+		setActiveIndexes((prev) => (prev.includes(index) ? prev.filter((i) => i !== index) : [...prev, index]))
+	}
+
+	return (
+		<div className='grid grid-cols-8 col-span-4 py-8 bg-white'>
+			<div className='col-start-3 col-span-4'>
+				<h1 className='text-[#00D56C] text-3xl font-bold'>{data.blockTitle}</h1>
+				<Accordion multiple>
+					{data.content?.map((item, index) => {
+						return item ? (
+							<Accordion.Item key={index} value={item.accordionTitle || ''}>
+								<Accordion.Control
+									icon={
+										activeIndexes.includes(index) ? (
+											<Iconify icon='mdi:minus' width={18} />
+										) : (
+											<Iconify icon='mdi:plus' width={18} />
+										)
+									}
+									onClick={() => toggleIndex(index)}
+								>
+									<p className='text-xl font-bold'>{item.accordionTitle}</p>
+								</Accordion.Control>
+								<Accordion.Panel>{item.accordionContent}</Accordion.Panel>
+							</Accordion.Item>
+						) : null
+					})}
+				</Accordion>
+			</div>
+		</div>
+	)
+}
+
+export const AccordionContainer = ({ data }: { data: PageBlocksAccordionblock }) => {
+	return <AccordionBlock data={data} />
+}
+
+export const accordionTemplate: Template = {
+	name: 'accordionblock',
```

Comment:
**NIT**: not an issue but i think we have been using snake case in the names of the components so I just noticed the difference here. Similarly, I saw your use of 'block' in the naming scheme for relative attributes and variables. I think it's better to stay away from the word 'block' directly here, especially since it can be confused in the context of _accordion block_ vs _accordion item_. I think just _accordion_ is good of a name itself.